pkg/ocp4/oauth/basicauth.go

```diff
@@ -44,14 +46,20 @@ func buildBasicAuthIP(serializer *json.Serializer, p configv1.IdentityProvider)
 	// TODO: Fetch cert and key
 	certFile := "456This is pretend content"
 	encoded := base64.StdEncoding.EncodeToString([]byte(certFile))
-	certSecret := secrets.GenSecret(certSecretName, encoded, "openshift-config", "basicauth")
+	certSecret, err := secrets.GenSecret(certSecretName, encoded, "openshift-config", "basicauth")
+	if err != nil {
+		return idP, *certSecret, *keySecret, err
```

Comment:
I would explicitly return nil for the `certSecret` and the `keySecret`. 
```golang
return idP, nil, nil, err
```
It may not work depending on the return types, but it looks weird to return `*certSecret`